### ISSUE TITLE
helmPublish task fails

### ISSUE THREAD

**Issue by jfarr** [unbroken-dome/gradle-helm-plugin#8]
The `helmPublish` task always fails with the following error:

```
Could not determine the dependencies of task ':helmPublishMainChartToExpRepo'.
> Expected configuration ':helmMain' to contain exactly one file, however, it contains no files.
```

What am I doing wrong?

**Environment**
```
LMTC-JFARR:repro jfarr1$ ./gradlew --version

------------------------------------------------------------
Gradle 5.1.1
------------------------------------------------------------

Build time:   2019-01-10 23:05:02 UTC
Revision:     3c9abb645fb83932c44e8610642393ad62116807

Kotlin DSL:   1.1.1
Kotlin:       1.3.11
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          1.8.0_171 (Oracle Corporation 25.171-b11)
OS:           Mac OS X 10.12.6 x86_64

LMTC-JFARR:repro jfarr1$ helm version
Client: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
```
**Repro**

[repro.zip](https://github.com/unbroken-dome/gradle-helm-plugin/files/2805572/repro.zip)

I get the same error with or without template files under `templates`, with or without running `helmInitClient helmPackage` first.

**Comment by wpbeckwith** [comment]
@jfarr I'm seeing the same thing.  Did you ever solve this?

**Comment by jfarr** [comment]
@wpbeckwith Sorry, I did not. I ended up working around it with the helm-push plugin. https://github.com/chartmuseum/helm-push

**Comment by wpbeckwith** [comment]
@tkrullmann Your plugin works great except for this one issue where we can't publish with it.  Would love some feedback from you or an example that use the plug-in so I can see what's different.  

Specifically, I'm seeing the following:

Caused by: java.lang.IllegalStateException: Expected configuration ':helmDevtest' to contain exactly one file, however, it contains no files.
        at org.gradle.api.internal.file.AbstractFileCollection.getSingleFile(AbstractFileCollection.java:61)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getSingleFile(Unknown Source)
        at org.unbrokendome.gradle.plugins.helm.publishing.rules.HelmPublishChartToRepositoryTaskRule$apply$$inlined$let$lambda$1$2.call(HelmPublishChartToRepositoryTaskRule.kt:52)
        at org.unbrokendome.gradle.plugins.helm.publishing.rules.HelmPublishChartToRepositoryTaskRule$apply$$inlined$let$lambda$1$2.call(HelmPublishChartToRepositoryTaskRule.kt:16)
        at org.gradle.api.internal.provider.DefaultProvider.getOrNull(DefaultProvider.java:41)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.isPresent(AbstractMinimalProvider.java:40)
        at org.gradle.api.internal.provider.AbstractMappingProvider.isPresent(AbstractMappingProvider.java:44)
        at org.gradle.api.internal.provider.DefaultPropertyState.isPresent(DefaultPropertyState.java:155)
        at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$DefaultPropertyValue.getValue(AbstractNestedRuntimeBeanNode.java:151)
        at org.gradle.api.internal.tasks.properties.bean.AbstractNestedRuntimeBeanNode$DefaultPropertyValue.call(AbstractNestedRuntimeBeanNode.java:160)
        at org.gradle.util.GUtil.uncheckedCall(GUtil.java:459)
        at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:64)
        at org.gradle.api.internal.tasks.TaskPropertyFileCollection.visitContents(TaskPropertyFileCollection.java:53)
        at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:186)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:106)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:211)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:66)
        ... 94 more

**Comment by tkrullmann** [comment]
This should be fixed now, please try with version 0.3.1.

The packaged chart artifact was resolved incorrectly from the rule that creates the publish task, it should be working now.
